statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -2894,6 +2894,85 @@ def zvalues(self):
         """
         return self.params / self.bse
 
+    def get_rsquared(self, baseline="rwdrift", **kwargs):
+        # calc sse finite estimation
+        d = np.maximum(self.loglikelihood_burn, self.nobs_diffuse)
+        srss = np.sum(self.standardized_forecasts_error[:, d:]**2, axis=1).T
+        f_T = np.array([self.forecasts_error_cov[i, i, -1]
+                       for i in range(self.model.k_endog)])
+        sse = f_T * srss
+        endog = self.model.endog
+
+        # calc rsquared wrt different baseline models
+        if baseline == "rwdrift":
+            ssdm = np.var(np.diff(endog, axis=0), axis=0) * (len(endog) - 1)
+            rsquared = 1. - sse / ssdm
+        elif baseline == "mean":
+            ssm = np.var(endog, axis=0) * len(endog)
+            rsquared = 1. - sse / ssm
+        elif baseline == "seasonal":
+            from statsmodels.regression.linear_model import OLS
+            from statsmodels.tools.tools import add_constant
```

Comment:
You could just use AutoReg which directly supports this model specification at a high level.

```
statsmodels.tsa.ar_model.AutoReg(endog, 0, trend='c', seasonal=True, periods=seasonal)
```